EventsExpress/ClientApp/src/components/inventory/InventoryList.js

```diff
@@ -48,28 +57,53 @@ class InventoryList extends Component {
 
     markItemAsEdit = inventar => {
         let updateList = this.props.inventories.items;
-        updateList.map(item => {
-            if (inventar.id === item.id)
-                item.isEdit = true;
+        updateList.find(e => e.id === inventar.id).isEdit = true;
+
+        this.props.get_inventories(updateList);
+        this.setState({
+            disabledEdit: true
         });
+    }
+
+    markItemAsWillTake = inventar => {
+        let updateList = this.props.inventories.items;
+        updateList.find(e => e.id === inventar.id).isWillTake = true;
+        updateList.find(e => e.id === inventar.id).isEdit = true;
 
         this.props.get_inventories(updateList);
         this.setState({
             disabledEdit: true
         });
     }
 
+    onAlreadyGet = inventar => {
+        let updateList = this.props.inventories.items;
+        updateList.map(item => {
+            if (inventar.id === item.id)
+                item.showAlreadyGetDetailed = !item.showAlreadyGetDetailed;
+        });
+
+        this.props.get_inventories(updateList);
```

Comment:
Why do you need to keep a global state of expanded items? Isn't it better to keep it in a local state of each displayed item? The less calls to redux - the better performance. Please consider moving the logic of expanding/collapsing details to display components to reduce render cycles.